core/controllers/skill_editor.py

```diff
@@ -198,11 +198,9 @@ def put(self, skill_id):
     def delete(self, skill_id):
         """Handles Delete requests."""
         skill_domain.Skill.require_valid_skill_id(skill_id)
-        skill_ids_assigned_to_some_topic = (
-            topic_services.get_all_skill_ids_assigned_to_some_topic())
-        if skill_id in skill_ids_assigned_to_some_topic:
-            raise self.InvalidInputException(
-                'Cannot delete skill that is assigned to a topic.')
+
+        skill_services.delete_skill_from_all_topics(self.user_id, skill_id)
```

Comment:
Sorry for the comment again, but it should have been `remove_skill_from_all_topics`.

Reply:
Oh no problem, its absolutely fine.
Done.